pkgs/development/interpreters/gnudatalanguage/default.nix

```
@@ -0,0 +1,167 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, wrapGAppsHook
+, readline
+, ncurses
+, zlib
+, gsl
+, openmp
+, graphicsmagick
+, netcdf
+, hdf4
+, hdf5
+, fftw
+, fftwFloat
+, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg
+# We enable it in hdf4 and use libtirpc as a dependency here from the passthru
+# of hdf4
+, enableLibtirpc ? stdenv.isLinux
+, libtirpc
+, python3
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+# Choose MPICH over OpenMPI because it currently builds on AArch and Darwin
+, mpi
+# Unfree optional dependency for hdf4 and hdf5
+, enableSzip ? false
+, szip
+# wxWidgets is preferred over X11 for this project but we only have it on Linux
+# and Darwin. Also, we use the wxWidgets dependency here from the passthru of
+# plplot.
+, enableWX ? (stdenv.isLinux || stdenv.isDarwin)
+# X11: OFF by default for platform consistency. Use X where WX is not available
+, enableXWin ? (!stdenv.isLinux && !stdenv.isDarwin)
+}:
+
+let
+  enablePlplotDrivers = enableWX || enableXWin;
+  hdf4-custom = hdf4.override {
+      uselibtirpc = enableLibtirpc;
+      szipSupport = enableSzip;
+      inherit szip;
+  };
+  hdf5-custom = hdf5.override {
+    mpiSupport = enableMPI;
+    inherit mpi;
+    szipSupport = enableSzip;
+    inherit szip;
+  };
+  netcdf-custom = netcdf.override {
+    hdf5 = hdf5-custom;
+  };
+  plplot-with-drivers = plplot.override {
+    inherit
+      enableWX
+      enableXWin
+    ;
+  };
+in stdenv.mkDerivation rec {
+  pname = "gnudatalanguage";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "gdl";
+    rev = "v${version}";
+    sha256 = "sha256-Y9LVRaWjQqpWqjNngxB406PE/rl/9S8rs0u0CK5ivUA=";
+  };
+
+  buildInputs = [
+    readline
+    ncurses
+    zlib
+    gsl
+    openmp
+    graphicsmagick
+    fftw fftwFloat fftwLongDouble
+    proj
+    shapelib
+    expat
+    mpi
+    udunits
+    eigen
+    pslib
+    eccodes
+    glpk
+    libpng
+    libtiff
+    libgeotiff
+    libjpeg
+    hdf4-custom hdf5-custom
+    plplot-with-drivers
+    netcdf-custom
+  ]
+  ++ lib.optional enableXWin plplot-with-drivers.libX11
+  ++ lib.optional enableWX plplot-with-drivers.wxWidgets
+  ++ lib.optional enableMPI mpi
+  ++ lib.optional enableLibtirpc hdf4-custom.libtirpc
+  ++ lib.optional enableSzip szip;
+
+  propagatedBuildInputs = [
+    (python3.withPackages (ps: with ps; [ numpy ]))
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ] ++ lib.optional enableWX wrapGAppsHook;
+
+  cmakeFlags =
+  lib.optional (!enablePlplotDrivers) "-DINTERACTIVE_GRAPHICS=OFF"
+  ++ lib.optional (!enableWX) "-DWXWIDGETS=OFF"
+  ++ lib.optional enableSzip "-DSZIPDIR=${szip}"
+  ++ lib.optionals enableXWin [ "-DX11=ON" "-DX11DIR=${plplot-with-drivers.libX11}" ]
+  ++ lib.optionals enableMPI [ "-DMPI=ON" "-DMPIDIR=${mpi}" ];
+
```

```suggestion
  cmakeFlags =
    lib.optional (!enablePlplotDrivers) "-DINTERACTIVE_GRAPHICS=OFF"
    ++ lib.optional (!enableWX) "-DWXWIDGETS=OFF"
    ++ lib.optional enableSzip "-DSZIPDIR=${szip}"
    ++ lib.optionals enableXWin [ "-DX11=ON" "-DX11DIR=${plplot-with-drivers.libX11}" ]
    ++ lib.optionals enableMPI [ "-DMPI=ON" "-DMPIDIR=${mpi}" ];

```
nixpkgs-fmt would do this